cluster_experiments/analytics/metric.py

```diff
@@ -0,0 +1,219 @@
+from abc import ABC, abstractmethod
+
+import pandas as pd
+
+
+class Metric(ABC):
+    """
+    An abstract base class used to represent a Metric with an alias.
+
+    Attributes
+    ----------
+    alias : str
+        A string representing the alias of the metric
+    """
+
+    def __init__(self, alias: str):
+        """
+        Parameters
+        ----------
+        alias : str
+            The alias of the metric
+        """
+        self._validate_alias(alias)
```

Comment:
My general opinion about inits is that the first thing they do should be setting parameters. Validations should come after, and then they are called without parameters, just self._validate_alias()